src/test/java/io/vertx/kafka/client/tests/ConsumerTestBase.java

```diff
@@ -1332,7 +1332,8 @@ public void testConsumerBatchHandler(TestContext ctx) throws Exception {
     String consumerId = topicName;
     Async batch1 = ctx.async();
     AtomicInteger index = new AtomicInteger();
-    int numMessages = 500;
```

Comment:
This resulted in a single iteration of the consumer loop, which effectively hid the NPE. I've bumped `numMessages` such that it actually triggers the issue.